src/yunohost/io.py

```diff
@@ -0,0 +1,222 @@
+    """
+    assert isinstance(file_path, str)
+
+    # Read file
+    file_content = _read_file(file_path)
+
+    # Try to load json to check if it's syntaxically correct
+    try:
+        loaded_json = json.loads(file_content)
+    except ValueError:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('io_corrupted_json', file=file_path))
+
+    return loaded_json
+
+
+def _write_json(file_path, input_dict):
+    """
+    Write a dictionnary to a json file
+
+    Keyword argument:
+        file_path -- Path to the json file
+        input_dict -- The dictionnary to write
+    """
+
+    # Assumptions
+    assert isinstance(file_path, str)
+    assert isinstance(input_dict, dict)
+
+    # Write dict to file
+    try:
+        with open(file_path, "w") as f:
+            json.dump(input_dict, f)
+    except IOError as e:
+        raise MoulinetteError(errno.EACCES,
+                              m18n.n('io_cannot_write_file', file=file_path))
+    except Exception as e:
+        raise MoulinetteError(errno.EIO,
+                              m18n.n('io_unknown_error_writing_file',
+                                     file=file_path, error=str(e)))
+
+
+def _remove_file(file_path):
+    """
+    Remove a regular file if it exists
+
+    Keyword argument:
+        file_path -- Path of the file to remove
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+
+    if os.path.exists(file_path):
+        try:
+            os.remove(file_path)
+        except Exception as e:
+            raise MoulinetteError(errno.EIO,
+                                  m18n.n('io_unknown_error_removing_file',
+                                         file=file_path, error=str(e)))
+
+
+def _set_permissions(file_path, user, group, permissions):
+    """
+    Change permissions of a given file.
+    Example : _set_permissions("/etc/swag", "root", "www-data", 0750)
+
+    Keyword argument:
+        file_path -- Path to the json file
+        user -- Some user name
+        group -- Some unix group name
+        permissions -- Permissions to set, preferrably in octal format
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+    assert isinstance(user, str)
+    assert isinstance(group, str)
+    assert isinstance(permissions, int)
+
+    # Check file exist
+    if not os.path.isfile(file_path):
+        raise MoulinetteError(errno.EROENT,
+                              m18n.n('io_no_such_file', file=file_path))
+
+    uid = pwd.getpwnam(user).pw_uid
+    gid = grp.getgrnam(group).gr_gid
+
+    os.chown(file_path, uid, gid)
+    os.chmod(file_path, permissions)
+
+
+def _download_text_data(url):
+    """
+    Download text from a url and returns the raw text
+
+    Keyword argument:
+        url -- The url to download the data from
+    """
+    # Assumptions
+    assert isinstance(file_path, str)
+
+    # Download file
+    try:
+        r = requests.get(url, timeout=30)
```

Comment:
According to requests documentation we should **always** have a timeout, so only the value should be modifiable http://docs.python-requests.org/en/latest/user/quickstart/#timeouts